src/libs/actions/TaxRate.ts

```diff
@@ -288,9 +289,9 @@ function deletePolicyTaxes(policyID: string, taxesToDelete: string[]) {
     const firstTaxID = Object.keys(policyTaxRates ?? {})
         .sort((a, b) => a.localeCompare(b))
         .at(0);
-    const customUnits = policy?.customUnits ?? {};
-    const customUnitID = Object.keys(customUnits).at(0) ?? '-1';
-    const ratesToUpdate = Object.values(customUnits?.[customUnitID]?.rates ?? {}).filter(
+    const distanceRateCustomUnit = PolicyUtils.getDistanceRateCustomUnit(policy);
+    const customUnitID = distanceRateCustomUnit?.customUnitID ?? '-1';
+    const ratesToUpdate = Object.values(distanceRateCustomUnit?.rates ?? {}).filter(
```

Comment:
Why are we using distance rate unit in taxes?

Reply:
This didn't answer my question. If we change a tax code then all the rates that are using that tax should use the new code (no matter which unit they belong to). IOW we shouldn't limit this to distance custom unit. Please revert

Reply:
Any rate could have attributes and could be using the tax attributes. The existing code is future proof and there is no need to make a change here.

Reply:
Any implemented (current or future) `CustomUnit` can have a rate that uses `attributes.taxRateExternalID` and when the used tax is updated/deleted that attribute should be changed as well. This is already handled with the existing code. The new code will cause a regression if any rate besides the distance rate got taxed.